src/query/list_file.c

```diff
@@ -392,6 +394,55 @@ qfile_list_cache_cleanup (THREAD_ENTRY * thread_p)
 }
 #endif
 
+int
+qcache_get_new_ht_no (THREAD_ENTRY * thread_p)
+{
+  int ht_no = -1;
+
+  if (QFILE_IS_LIST_CACHE_DISABLED)
+    {
+      return NO_ERROR;
+    }
+
+  if (csect_enter (thread_p, CSECT_QPROC_LIST_CACHE, INF_WAIT) != NO_ERROR)
+    {
+      return ER_FAILED;
+    }
+
+  if (qfile_List_cache.next >= 0)
```

Comment:
It looks good to write 'if (qfile_List_cache.next >= 0)' before csect_enter() .

Reply:
Ok, I will fix it.